be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/validators/MeetingValidator.scala

```diff
@@ -39,23 +39,39 @@ sealed class MeetingValidator(dbActorRef: => AskableActorRef) extends MessageDat
         val sender: PublicKey = message.sender
         val channel: Channel = rpcMessage.getParamsChannel
 
-        if (!validateTimestampStaleness(data.creation)) {
-          Right(validationError(s"stale 'creation' timestamp (${data.creation})"))
-        } else if (!validateTimestampStaleness(data.start)) {
-          Right(validationError(s"stale 'start' timestamp (${data.start})"))
-        } else if (data.end.isDefined && !validateTimestampOrder(data.creation, data.end.get)) {
-          Right(validationError(s"'end' (${data.end.get}) timestamp is smaller than 'creation' (${data.creation})"))
-        } else if (data.end.isDefined && !validateTimestampOrder(data.start, data.end.get)) {
-          Right(validationError(s"'end' (${data.end.get}) timestamp is smaller than 'start' (${data.start})"))
-        } else if (expectedHash != data.id) {
-          Right(validationError("unexpected id"))
-        } else if (!validateOwner(sender, channel, dbActorRef)) {
-          Right(validationError(s"invalid sender $sender"))
-        } else if (!validateChannelType(ObjectType.LAO, channel, dbActorRef)) {
-          Right(validationError(s"trying to send an CreateMeeting message on a wrong type of channel $channel"))
-        } else {
-          Left(rpcMessage)
-        }
+        runChecks(
+          checkTimestampStaleness(
+            rpcMessage,
+            data.creation,
+            validationError(s"stale 'creation' timestamp (${data.creation})")
+          ),
+          checkTimestampStaleness(
+            rpcMessage,
+            data.start,
+            validationError(s"stale 'start' timestamp (${data.start})")
+          ),
+          checkTimestampOrder(
+            rpcMessage,
+            data.creation,
+            data.end.get,
+            validationError(s"'end' (${data.end.get}) timestamp is smaller than 'creation' (${data.creation})")
+          ),
+          checkTimestampOrder(
+            rpcMessage,
+            data.start,
+            data.end.get,
```

Comment:
Same as above, I think that line could be problematic, if data.end is undefined an exception will be thrown instead of using the error pipeline.

Reply:
Here you can use your function `CheckOptionalTimestampOrder`.